query-coordinator/src/test/scala/com/socrata/querycoordinator/QueryRewritingTestUtility.scala

```diff
@@ -45,7 +46,17 @@ object QueryRewritingTestUtility {
       (a: AnalyzedDatasetContext) => (b: ParsedSoql) => analyzer.analyzeBinary(b)(a),
       // aka ReMap
       (a: ColumnMappings) => (b: AnalyzedSoql) => QueryParser.remapAnalyses(a, b),
-      (a, b) => rewriter.possibleRewrites(a, b, false)
+      (a, b) => {
+        a match {
+          case Leaf(analysis) => possiblyRewriteQuery(rewriter, analysis, QueryRewriter.mergeRollupsAnalysis(b)) match {
```

Comment:
Try to route the rewriter down the most correct path.
We have a "regular" for leafs, and a "compound" for piped/compound..etc.